src/download.py

```diff
@@ -71,29 +91,25 @@ async def get_remote_file_size(self, url: str | URL) -> "tuple[URL, int]":
     async def download_chunk(
         self, url: str | URL, start: int, end: int, buffer_view: memoryview
     ) -> None:
-        for i in range(5):
-            headers = {"Retry-Count": str(i)} if i else {}
-            try:
-                headers |= {"Range": f"bytes={start}-{end}"}
-                async with self.session.get(url, headers=headers) as response:
-                    buffer_view[start : end + 1] = await response.read()
-                    return
-            except (aiohttp.ClientError, asyncio.TimeoutError) as e:
-                print(f"Error: {e}")
-                await asyncio.sleep(random.random() / 10)  # sleep 0-100ms
+        async with self.sem:
+            for i in range(5):
+                headers = {"Retry-Count": str(i)} if i else {}
+                try:
+                    headers |= {"Range": f"bytes={start}-{end}"}
+                    async with self.session.get(url, headers=headers) as response:
```

Comment:
I'm probably going to say something here that adds a lot of complexity: If we are using http/2 (I have not looked into requests support and if we are here), we should endeavor to reuse connections to the same upstream server. If this it still http 1.x 🤷 

pget does not do this, but it likely should.

Reply:
alas, [aiohttp does not support http/2](https://github.com/aio-libs/aiohttp/issues/5631), but thankfully 1.1 still supports pipelining which is the important part, not having multiplexing is not ideal but tolerable